kratos/geometries/geometry.h

```diff
@@ -3169,6 +3192,42 @@ class Geometry
 
     PointsArrayType mPoints;
 
+    ///@}
+    ///@name Id Bit-Change Operations
+    ///@{
+
+    /// Checks first bit in mId. 0 -> id; 1 -> name
+    bool IsGeometryIdString()
+    {
+        return mId & (IndexType(1) << 63);
+    }
+
+    inline bool IsGeometryIdString(IndexType Id) const
+    {
+        return Id & (IndexType(1) << 63);
+    }
+
+    void SetGeometryIdString(IndexType& Id)
+    {
+        Id |= (IndexType(1) << 63);
+    }
+
+    /// Checks second bit in mId. 0 -> foreign id; 1 -> self assigned
+    bool IsSelfAssignedId()
+    {
+        return mId & (IndexType(1) << 62);
+    }
+
+    inline bool IsSelfAssignedId(IndexType Id) const
+    {
+        return mId & (IndexType(1) << 62);
+    }
+
+    void SetSelfAssignedId(IndexType& Id) const
+    {
+        Id |= (IndexType(1) << 62);
```

Comment:
I'm not sure if this is correct. I think you are setting to automatically assigned.

Reply:
I think it should be correct. As soon as this function is called the second bit is set to be 1, which actually means that it is self assigned.

Reply:
Now I see, you are changing the argument because you set it afterward.